helpers/helpers_emba_sbom_helpers.sh

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash -p
+
+# EMBA - EMBEDDED LINUX ANALYZER
+#
+# Copyright 2024-2024 Siemens Energy AG
+#
+# EMBA comes with ABSOLUTELY NO WARRANTY. This is free software, and you are
+# welcome to redistribute it under the terms of the GNU General Public License.
+# See LICENSE file for usage of this software.
+#
+# EMBA is licensed under GPLv3
+# SPDX-License-Identifier: GPL-3.0-only
+#
+# Author(s): Michael Messner
+
+# Description: Helper functions for SBOM building
+#
+
+# first: build the properaties path array
+# This can be used for the binary path (souce_path) and for paths extracted from a
+# package like deb or rpm (path). Additionally, it is commonly used for the architecture
+# and further meta data like the version identifier
+# parameter: array with all the properties in the form
+#   "path:the_path_to_log"
+#   "other_propertie:the_property_to_log"
+# returns global array PROPERTIES_JSON_ARR
+build_sbom_json_properties_arr() {
+  local lPROPERTIES_ARRAY_INIT_ARR=("$@")
+
+  local lPROPERTIES_ELEMENT_ID=""
+  local lPROPERTIES_ELEMENT=""
+  # PROPERTIES_JSON_ARR is used in the caller
+  export PROPERTIES_JSON_ARR=()
+  local lPROPERTIES_ELEMENT_1=""
+  local lPROPERTIES_ELEMENT_2=""
+  local lINIT_ELEMENT=""
+
+  for lPROPERTIES_ELEMENT_ID in "${!lPROPERTIES_ARRAY_INIT_ARR[@]}"; do
+    lPROPERTIES_ELEMENT="${lPROPERTIES_ARRAY_INIT_ARR["${lPROPERTIES_ELEMENT_ID}"]}"
+    # lPROPERTIES_ELEMENT_1 -> path, source or something else
+    lPROPERTIES_ELEMENT_1=$(echo "${lPROPERTIES_ELEMENT}" | cut -d ':' -f1)
+    # lPROPERTIES_ELEMENT_2 -> the real value
+    lPROPERTIES_ELEMENT_2=$(echo "${lPROPERTIES_ELEMENT}" | cut -d ':' -f2-)
+    # jo is looking for a file if our entry starts with : -> lets pseudo escape it for jo
+    if [[ "${lPROPERTIES_ELEMENT_2:0:1}" == ":" ]]; then
+      # shellcheck disable=SC1003
+      lPROPERTIES_ELEMENT_2='\'"${lPROPERTIES_ELEMENT_2}"
```

Comment:
Unsure what this actually does
Shouldn't adding a \ be \\ ?

Reply:
if we have a ":" as first character we need to escape it for jo with a '\'